comp/metadata/inventoryagent/inventoryagent.go

```diff
@@ -60,19 +60,22 @@ func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
 type inventoryagent struct {
 	util.InventoryPayload
 
-	log      log.Component
-	conf     config.Component
-	m        sync.Mutex
-	data     agentMetadata
-	hostname string
+	log          log.Component
+	conf         config.Component
+	sysprobeConf sysprobeconfig.Component
+	m            sync.Mutex
+	data         agentMetadata
+	hostname     string
 }
 
 type dependencies struct {
 	fx.In
 
-	Log        log.Component
-	Config     config.Component
-	Serializer serializer.MetricSerializer
+	Log    log.Component
+	Config config.Component
+	// if sysprobeconfig was loaded we use it, if not the system probe metadata will default to false.
+	SysProbeConfig sysprobeconfig.Component `optional:"true"`
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Might be a good candidate for `fxoptional.Component` https://github.com/DataDog/datadog-agent/pull/21625

Reply:
I don't think so, we want to get `sysprobeconfig` if it was loaded. We don't want to force user to load an empty `sysprobeconfig` everywhere the metadata are used.

Reply:
Our internal FAQ strongly discourages the usage of `optional:"true"`.

Using [optional:"true"](https://pkg.go.dev/go.uber.org/fx#hdr-Optional_Dependencies) should be discouraged because it can hide errors. For example consider [this code](https://go.dev/play/p/xrzexhbCeif) where an user can expected to get an error because the dependency `FooDep` for `CreateFoo` is missing but the code doesn't raise an error. See also https://github.com/uber-go/fx/issues/613.

Is there something new that justify changing this guideline?
cc @truthbk @gh123man as we discussed this topic.

I would like to emphase that using `optional:"true"` prevents catching missing dependencies. It is like removing tests on subcommands like [this one](https://github.com/DataDog/datadog-agent/blob/7.50.x/cmd/agent/subcommands/run/command_test.go#L21). These kind of tests are very useful as there were two binaries without this kind of tests and we had missing dependencies on these two binaries.

> We don't want to force user to load an empty sysprobeconfig everywhere the metadata are used.

Can you elaborate on why this is different from the other cases? Is it to avoid bringing `SysProbeConfig` dependencies (If yes there are easy solution to solve that).